beacon-chain/blockchain/van_process_pending_blocks.go

```diff
@@ -220,16 +225,43 @@ func (s *Service) waitForConfirmation(b interfaces.SignedBeaconBlock) error {
 	}
 }
 
-// verifyPandoraShardInfo
+// verifyPandoraShardInfo verifies Pandora shard info
 func (s *Service) verifyPandoraShardInfo(parentBlk, curBlk interfaces.SignedBeaconBlock) error {
+	var parentPanShards []*ethpb.PandoraShard
+
+	// Current beacon block
+	if curBlk == nil || curBlk.IsNil() {
+		return errUnknownCurrent
+	}
+	curBlkObj := curBlk.Block()
```

Comment:
We do not need to check block because if you see IsNil() method, you can see this method already checks this

Reply:
@blazejkrzak have you checked `IsNil()` method?

Reply:
@atif-konasl There are only option that somehow someone will give me `nil` interface. So because `IsNil()` is checking only `w.b == nil || w.Body().IsNil()` not the interface `w` then I've just secured this too.

Reply:
Yes, I have checked. In a matter of fact its quite simple. You cannot invoke any method on nil object. You cannot do `(nil)(*ethpb.SignedBlock).isNil()`

Reply:
you will get panic

Reply:
Removing `nil` checks would be possible, if those functions returned non-nilable objects (for example structs). With pointers and interfaces, if crashing is not the desired way of handling errors, we need to check it all

Reply:
@atif-konasl So as we discussed, because of `prysmaticlabs` implementation I should remove 
```
curBlkObjBody := curBlkObj.Body()
if curBlkObjBody.IsNil() {
	return errUnknownCurrentBody
}
````
 because ` curBlk.IsNil()` checks for  `curBlk.Block().isNil()` and `curBlk.Block().Body().isNil()` and just provide `curPanShards := curBlk.Block().Body().PandoraShards()`, right?

Reply:
@atif-konasl I solved this by providing above implementation changes. You can re-review now.